doc/architecture/001-document-decisions.md

```diff
@@ -0,0 +1,5 @@
+# Document Decisions
+
+## Context
+
+The purpose of this directory currently, is to document architectural decisions discussed for future changes.
```

Comment:
Should we keep this document tethered to the current code conditions only? I.e., describes what the code does right now, without historical or future contexts. I think that is a good compromise. It keeps things shorter (more maintainable) and doesn't dilute anything for the new developer (e.g., addingin in additional contexts).